### PR TITLE
Wrap delete all versions into anonymous function

### PR DESCRIPTION
**Pull request by darrell-d**
## Checklist

- [x] unit tests added and/or verified that tests pass
- [x] I have considered any possible security implications of this change
- [x] I have considered deployment issues.

### CODE REVIEW

File: jobs/src/main/scala/com/pennsieve/jobs/types/DeleteJob.scala [hunk 1]
```diff
@@ -457,15 +457,18 @@ class DeleteJob(
         )
         val req = new DeleteObjectsRequest(bucket)
           .withKeys(s3Keys: _*)
-        try {
-          s3Keys.foreach { key =>
+        def deleteVersionsForKeys(keys: Seq[String]): Unit = {
+          keys.foreach { key =>
             deleteAllVersions(bucket, key, amazonS3)
           }
+        }
```
Comment (edmore) [inline jobs/src/main/scala/com/pennsieve/jobs/types/DeleteJob.scala:464]: you could perhaps move this out :

```
  def deleteVersionsForKeys(bucket: String, keys: Seq[String]): Unit = {
    keys.foreach { key =>
      deleteAllVersions(bucket, key, amazonS3)
    }
  }
```

File: jobs/src/main/scala/com/pennsieve/jobs/types/DeleteJob.scala [hunk 2]
```diff
@@ -475,6 +478,7 @@ class DeleteJob(
           case ex: MultiObjectDeleteException =>
             val deletedIds = ex.getDeletedObjects.asScala
               .map(_.getKey)
+            deleteVersionsForKeys(s3Keys)
```
Comment (edmore) [inline jobs/src/main/scala/com/pennsieve/jobs/types/DeleteJob.scala:481]: what is the difference between the s3Keys and deletedKeys (deletedIds.toSeq) in this particular case?

On failure should we be doing the same thing we are doing in the `try` segment of the code?

Comment (darrell-d) [reply jobs/src/main/scala/com/pennsieve/jobs/types/DeleteJob.scala]: The S3keys are a string path to where the file lives, the deletedIds.toSeq is an ID result from the delete.

RE on failure block: yeah let me remove that

**Comment by muftring** [comment]
looks ok to me, with one question/comment: 

what is the effect of moving when `deleteAllVersions()` is called? Prior to this change it was before the call to `amazonS3.deleteObjects()`, with this change it is now after.

**Comment by darrell-d** [comment]
> looks ok to me, with one question/comment:
> 
> what is the effect of moving when `deleteAllVersions()` is called? Prior to this change it was before the call to `amazonS3.deleteObjects()`, with this change it is now after.

Previously when this ran, the versions were being deleted, but not the delete marker, which should be another version. 

This change is a step towards doing things in the correct order in an effort to correct this behavior: first delete then file, then sweep away all versions, as the deleteAllVersions function specifically looks to get a list of versions of the file.

**Review by edmore — APPROVED** [review]

**Pull request merged**